File: python-pr/52/python/code-quality-readability-maintainability/database_handler_13.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def handle_db_operation(operation, table, data=None, condition=None):
    if operation == 'select':
        if condition:
            return f"SELECT * FROM {table} WHERE {condition}"
        else:
            return f"SELECT * FROM {table}"
    elif operation == 'insert':
        if data:
            columns = ', '.join(data.keys())
            values = ', '.join([f"'{v}'" for v in data.values()])
            return f"INSERT INTO {table} ({columns}) VALUES ({values})"
    elif operation == 'update':
        if data and condition:
            set_clause = ', '.join([f"{k}='{v}'" for k, v in data.items()])
            return f"UPDATE {table} SET {set_clause} WHERE {condition}"
    elif operation == 'delete':
        if condition:
            return f"DELETE FROM {table} WHERE {condition}"
    return None
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/database_handler_13.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function handleDbOperation(operation, table, data = null, condition = null) {
    if (operation === 'select') {
        return condition ? `SELECT * FROM ${table} WHERE ${condition}` : `SELECT * FROM ${table}`;
    } else if (operation === 'insert') {
        if (data) {
            let columns = Object.keys(data).join(', ');
            let values = Object.values(data).map(v => `'${v}'`).join(', ');
            return `INSERT INTO ${table} (${columns}) VALUES (${values})`;
        }
    } else if (operation === 'update') {
        if (data && condition) {
            let setClause = Object.entries(data).map(([k, v]) => `${k}='${v}'`).join(', ');
            return `UPDATE ${table} SET ${setClause} WHERE ${condition}`;
        }
    } else if (operation === 'delete') {
        return condition ? `DELETE FROM ${table} WHERE ${condition}` : null;
    }
    return null;
}
// {/fact}